### DISCUSSION TITLE
about the value of !gdl.release

### DISCUSSION THREAD

**Discussion by alaingdl** [gnudatalanguage/gdl#1775] [General]
It is smart that we have now an automatic increment of the versioning, and also some tags related to the git version ...

But I would like also that we follow a convention concerning the !gdl.release content, like X.Y.Z or X.Y.Z-git
with X, Y, Z are numbers between 0 and 99 ! (and my stupid code  GDL_VERSION() can continue to work !)

```
GDL> print, !gdl.release
v1.0.2-538-gebee9df7
IDL> print,!version.release
8.8.0
FL> print,!fl.release
0.79.53
```

Comments on that point are welcome ! Especially @GillesDuvert @slayoo 
(please note that those fields are used since years in the benchmark :smile: )

**Comment by alaingdl** [comment]
but I am OK to put that in the git ...
```
; extra processing for GDL
if GDL_IDL_FL() EQ 'GDL' then begin
   if (STRUPCASE(STRMID(suite,0,1)) EQ 'V') then suite=STRMID(suite,1)
   if (STRPOS(suite, '-') GT 0) then suite=STRMID(suite,0,STRPOS(suite, '-')-1)
endif
```

**Comment by GillesDuvert** [comment]
I have no definitive opinion...
On the "compatibility side",  only `!version.release` MUST be considererd, and for GDL it is a 'fake' version, fixed to 8.2 but changeable to X.y with option `--fakerelease` . This to accomodate very old code (and perhaps improperly, at that).
`!gdl.release` should IMHO be something like `1.0.5` for releases, and anything like `v1.0.2-538-gebee9df7` for nightly builds etc. the latter permits to identify more clearly that it is an 'intermediate' build with the corresponding merge id.

But yes, indeed, a modified GDL_VERSION() seems inevitable.

**Comment by slayoo** [comment]
So perhaps, should we remove the "v" prefix when setting `!gdl.release`? Then, tagged releases will yield things like "1.0.5" there, while custom builds from git will have additional suffix like "1.0.5-10-xxxxxxxxx"

**Comment by slayoo** [comment]
here's a PR: https://github.com/gnudatalanguage/gdl/pull/1780

**Comment by alaingdl** [comment]
OK, back on this versioning issue. I spend some time playing with git related commands on 3 computers,
in all cases I have, on a fresh `git clone https://github.com/gnudatalanguage/gdl` and old version number : 

```
$ git describe
v1.0.0-JOSS-830-g86806a3b
$ git tag
v0.8.10
v0.8.11
v0.8.9
v0.9
v0.9.1
[...]
v1.0.0-JOSS
[...]
v1.0.1
v1.0.2
v1.0.3
v1.0.4
weekly-release
```

But when I compile the up-to-date code, I have : 
```
  GDL - GNU Data Language, Version v1.0.2-545-g86806a3b-dirty
GDL> !gdl
{
    "RELEASE": "v1.0.2-545-g86806a3b-dirty",

```

I see now way to move into a apparent 1.0.4 something  ...

**Reply by slayoo** [reply]
@alaingdl, thanks for checking. This seems surprising. Locally, just trying the following commands:
```bash
git clone --recursive git@github.com:gnudatalanguage/gdl.git
cd gdl
mkdir build
cd build
cmake .. -DWXWIDGETS=OFF -DMAGICK=OFF -DTIFF=OFF -DNETCDF=OFF -DHDF=OFF -DHDF5=OFF -DFFTW=OFF -DLIBPROJ=OFF -DUDUNITS2=OFF -DEIGEN3=OFF -DGRIB=OFF -DGLPK=OFF
make -j 3
./src/gdl -quiet -e 'print, !gdl.release'
```
I get this as output from the last line:
```
1.0.2-546-g93ca3eab
```
If I checkout the v1.0.4 tag, rebuild, and try printing again:
```bash
git checkout v1.0.4
make -j 3
./src/gdl -quiet -e 'print, !gdl.release'
```
the output is:
```
1.0.4
```

How the above commands differ from how you compile GDL?
It should definitely not be `-dirty` on a fresh clone.
